modules/cookbooks/pages/rag.adoc

```diff
@@ -0,0 +1,174 @@
+    consumer_group: rp_connect_articles_group
+    tls:
+      enabled: true
+    sasl:
+      mechanism: SCRAM-SHA-256
+      user: "${REDPANDA_USER}"
+      password: "${REDPANDA_PASSWORD}"
+----
+
+We'll be using https://ollama.com/library/nomic-embed-text[Nomic Embed] to compute embeddings. Since each request only applies to a single document we will make this scale by making requests in parallel across our document batches.
+
+In order to send a mapped request and map the response back into the original document we will use the xref:components:processors/branch.adoc[`branch` processor], with a child xref:components:processors/ollama_embeddings.adoc[`ollama_embeddings`] processor.
+
+[source,yaml]
+----
+pipeline:
+  threads: -1
+  processors:
+    - branch:
+        request_map: 'root = "search_document: %s\n%s".format(this.article.title, this.article.content)'
+        processors:
+          - ollama_embeddings:
+              model: nomic-embed-text
+        result_map: 'root.article.embeddings = this'
+----
+
+With this pipeline our documents will come out looking something like this:
+
+[source,yaml]
+----
+{
+  "type": "article",
+  "article": {
+    "id": "123foo",
+    "title": "Dogs Stop Barking",
+    "content": "The world was shocked this morning to find that all dogs have stopped barking.",
+    "embeddings": [0.754, 0.19283, 0.231, 0.834], # This vector will actually have 768 dimensions
+  }
+}
+----
+
+Now we can send that to Postgres using the xref:components:outputs/sql_insert.adoc[`sql_insert`] output. We'll also take advantage of the `init_statement` functionality to setup our `pgvector` and our table.
+
+[source,yaml]
+----
+output:
+  sql_insert:
+    driver: postgres
+    dsn: "${PG_CONNECTION_STRING}"
+    init_statement: |
+      CREATE EXTENSION IF NOT EXISTS vector;
+      CREATE TABLE IF NOT EXISTS searchable_text (
+        id varchar(128) PRIMARY KEY,
+        title text NOT NULL,
+        body text NOT NULL,
+        embeddings vector(768) NOT NULL
+      );
+      CREATE INDEX IF NOT EXISTS text_hnsw_index
+        ON searchable_text 
+        USING hnsw (embeddings vector_l2_ops);
+    table: searchable_text
+    columns: ["id", "title", "body", "embeddings"]
+    args_mapping: "[this.article.id, this.article.title, this.article.content, this.article.embeddings.vector()]"
+----
+
+We can save this pipeline as `indexing.yaml` and run it with `rpk connect run ./indexing.yaml` to test and make sure our postgres table is being populated with embeddings.
+
+== Generating search responses
+
+In order to generate responses to questions over our dataset we're creating embeddings for, we'll use an xref:components:inputs/http_server.adoc[HTTP server input] to recieve the questions. Since we'll be running this pipeline in Streams mode, our HTTP server will xref:guides:streams_mode/about.adoc#http-endpoints[endpoints prefixed with stream identifier], which in our case is the same as the filename: `search`.
```

Comment:
```suggestion
To generate responses to questions with the dataset you're creating embeddings for, use a xref:components:inputs/http_server.adoc[HTTP server input] to receive the questions. Since you're running this pipeline in streams mode, the HTTP server uses xref:guides:streams_mode/about.adoc#http-endpoints[endpoints prefixed with stream identifier], which in this case is the same as the filename: `search`.
```